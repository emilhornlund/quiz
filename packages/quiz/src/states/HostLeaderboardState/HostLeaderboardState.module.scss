@import '../../styles/colors';
@import '../../styles/helpers';

.main {
  display: flex;
  flex-direction: column;
  height: 100%;
  flex: 1;

  @include mobile {
    row-gap: $layout-gutter;
    width: 100%;
  }

  @include tablet {
    row-gap: $layout-gutter * 1.5;
    width: $mobile-width;
  }

  @include desktop {
    row-gap: $layout-gutter * 2;
    width: $tablet-width;
  }

  .title {
    display: flex;
    flex-direction: row;
    justify-content: center;
    font-weight: bold;
    text-align: center;
    color: white;
    width: 100%;

    @include mobile {
      @include fontSize(18px);
    }

    @include tablet {
      @include fontSize(28px);
    }

    @include desktop {
      @include fontSize(38px);
    }
  }

  .leaderboard {
    display: grid;
    grid-template-columns: min-content 1fr min-content;
    row-gap: 0.5rem;
    width: 100%;

    .column {
      display: flex;
      flex-direction: column;
      justify-content: center;
      background-color: $white;
      border-radius: 0;
      color: rgba(0, 0, 0, 0.7);

      @include mobile {
        @include fontSize(16px);

        padding: $layout-gutter * 0.5;
      }

      @include tablet {
        @include fontSize(22px);

        padding: $layout-gutter * 0.75;
      }

      @include desktop {
        @include fontSize(28px);

        padding: $layout-gutter;
      }

      &:nth-child(-n+3) {
        background-color: $gold;
      }

      &:nth-child(n+4):nth-child(-n+6) {
        background-color: $silver;
      }

      &:nth-child(n+7):nth-child(-n+9) {
        background-color: $bronze;
      }

      &:nth-child(3n+1) {
        @include mobile {
          border-top-left-radius: $layout-gutter * 0.5;
          border-bottom-left-radius: $layout-gutter * 0.5;
        }

        @include tablet {
          border-top-left-radius: $layout-gutter * 0.75;
          border-bottom-left-radius: $layout-gutter * 0.75;
        }

        @include desktop {
          border-top-left-radius: $layout-gutter;
          border-bottom-left-radius: $layout-gutter;
        }
      }

      &:nth-child(3n+3) {
        align-items: flex-end;
        font-weight: bold;

        @include mobile {
          border-top-right-radius: $layout-gutter * 0.5;
          border-bottom-right-radius: $layout-gutter * 0.5;
        }

        @include tablet {
          border-top-right-radius: $layout-gutter * 0.75;
          border-bottom-right-radius: $layout-gutter * 0.75;
        }

        @include desktop {
          border-top-right-radius: $layout-gutter;
          border-bottom-right-radius: $layout-gutter;
        }
      }

      .position {
        display: flex;
        align-items: center;
        justify-content: center;

        span {
          display: flex;
          align-items: center;
          justify-content: center;
          aspect-ratio: 1 / 1;
          height: 100%;
          border-radius: 50%;
          background-color: rgba(0, 0, 0, 0.2);
          color: rgba(0, 0, 0, 0.7);
          font-weight: bold;

          @include mobile {
            padding: 0.2rem;
            @include fontSize(16px);
          }

          @include tablet {
            padding: 0.2rem;
            @include fontSize(22px);
          }

          @include desktop {
            padding: 0.2rem;
            @include fontSize(28px);
          }
        }
      }

      .score {
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: rgba(0, 0, 0, 0.1);
        color: rgba(0, 0, 0, 0.8);

        @include mobile {
          padding: 0 $layout-gutter * 0.5;
          border-radius: pxToRem(8px);
        }

        @include tablet {
          padding: 0 $layout-gutter * 0.75;
          border-radius: pxToRem(10px);
        }

        @include desktop {
          padding: 0 $layout-gutter;
          border-radius: pxToRem(12px);
        }
      }
    }
  }
}