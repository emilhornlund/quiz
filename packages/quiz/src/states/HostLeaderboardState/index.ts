export type { HostLeaderboardStateProps } from './HostLeaderboardState'
export { default } from './HostLeaderboardState'
