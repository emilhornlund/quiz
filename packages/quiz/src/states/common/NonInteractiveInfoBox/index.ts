export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export { default } from './NonInteractiveInfoBox'
