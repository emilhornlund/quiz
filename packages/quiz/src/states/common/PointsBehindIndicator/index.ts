export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export { default } from './PointsBehindIndicator'
