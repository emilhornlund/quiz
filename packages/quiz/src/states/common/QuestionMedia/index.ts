export type { QuestionMediaProps } from './QuestionMedia'
export { default } from './QuestionMedia'
