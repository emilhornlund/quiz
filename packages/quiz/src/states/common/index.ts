export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export type { QuestionAnswerPickerProps } from './QuestionAnswerPicker'
export type { QuestionMediaProps } from './QuestionMedia'
export type { QuestionTextPreviewProps } from './QuestionTextPreview'
export type { QuestionTypePointsBarProps } from './QuestionTypePointsBar'

export { default as GamePage } from './GamePage'
export { default as NonInteractiveInfoBox } from './NonInteractiveInfoBox'
export { default as PointsBehindIndicator } from './PointsBehindIndicator'
export { default as QuestionAnswerPicker } from './QuestionAnswerPicker'
export { default as QuestionMedia } from './QuestionMedia'
export { default as QuestionTextPreview } from './QuestionTextPreview'
export { default as QuestionTypePointsBar } from './QuestionTypePointsBar'
