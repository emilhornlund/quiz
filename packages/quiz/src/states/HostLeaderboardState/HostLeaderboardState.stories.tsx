import { GameEventType } from '@quiz/common'
import type { Meta, StoryObj } from '@storybook/react'

import HostLeaderboardState from './HostLeaderboardState'

const meta = {
  component: HostLeaderboardState,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof HostLeaderboardState>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {
    event: {
      type: GameEventType.LeaderboardHost,
      gamePIN: '123456',
      leaderboard: [
        { position: 1, nickname: 'ShadowCyborg', score: 18456, streaks: 2 },
        { position: 2, nickname: 'Radar', score: 18398, streaks: 0 },
        { position: 3, nickname: 'ShadowWhirlwind', score: 15492, streaks: 0 },
        { position: 4, nickname: 'WhiskerFox', score: 14118, streaks: 5 },
        { position: 5, nickname: 'JollyNimbus', score: 13463, streaks: 0 },
        { position: 6, nickname: 'PuddingPop', score: 12459, streaks: 0 },
        { position: 7, nickname: 'MysticPine', score: 11086, streaks: 2 },
        { position: 8, nickname: 'FrostyBear', score: 10361, streaks: 1 },
        { position: 9, nickname: 'Willo', score: 9360, streaks: 0 },
        { position: 10, nickname: 'ScarletFlame', score: 6723, streaks: 0 },
      ],
      pagination: { current: 1, total: 20 },
    },
  },
} satisfies Story
