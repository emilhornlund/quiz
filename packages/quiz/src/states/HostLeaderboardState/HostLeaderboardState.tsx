import { GameEventLeaderboardHost } from '@quiz/common'
import React, { FC } from 'react'

import { Button, Page } from '../../components'

import styles from './HostLeaderboardState.module.scss'

export interface HostLeaderboardStateProps {
  event: GameEventLeaderboardHost
}

const HostLeaderboardState: FC<HostLeaderboardStateProps> = ({
  event: { leaderboard },
}) => {
  return (
    <Page
      header={
        <Button
          id={'next-button'}
          type="button"
          kind="secondary"
          size="small"
          value="Next"
          arrow="right"
        />
      }
      noPadding>
      <div className={styles.main}>
        <div className={styles.title}>Leaderboard</div>
        <div className={styles.leaderboard}>
          {leaderboard.map(({ position, nickname, score }) => (
            <>
              <div key={`position_${position}`} className={styles.column}>
                <div className={styles.position}>
                  <span>{position}</span>
                </div>
              </div>
              <div key={`nickname_${position}`} className={styles.column}>
                {nickname}
              </div>
              <div key={`score_${position}`} className={styles.column}>
                <span className={styles.score}>{score}</span>
              </div>
            </>
          ))}
        </div>
      </div>
    </Page>
  )
}

export default HostLeaderboardState
