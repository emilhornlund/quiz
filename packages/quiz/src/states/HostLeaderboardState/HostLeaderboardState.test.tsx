import { GameEventType } from '@quiz/common'
import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import HostLeaderboardState from './HostLeaderboardState'

describe('HostLeaderboardState', () => {
  it('should render HostLeaderboardState', async () => {
    const { container } = render(
      <HostLeaderboardState
        event={{
          type: GameEventType.LeaderboardHost,
          gamePIN: '123456',
          leaderboard: [
            { position: 1, nickname: 'ShadowCyborg', score: 18456 },
            { position: 2, nickname: 'Radar', score: 18398 },
            { position: 3, nickname: 'ShadowWhirlwind', score: 15492 },
            { position: 4, nickname: 'WhiskerFox', score: 14118 },
            { position: 5, nickname: 'JollyNimbus', score: 13463 },
            { position: 6, nickname: 'PuddingPop', score: 12459 },
            { position: 7, nickname: 'MysticPine', score: 11086 },
            { position: 8, nickname: 'FrostyBear', score: 10361 },
            { position: 9, nickname: 'Willo', score: 9360 },
            { position: 10, nickname: 'ScarletFlame', score: 6723 },
          ],
          question: { current: 1, total: 20 },
        }}
      />,
    )

    expect(container).toMatchSnapshot()
  })
})
