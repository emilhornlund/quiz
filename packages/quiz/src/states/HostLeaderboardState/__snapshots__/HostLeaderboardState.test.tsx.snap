// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HostLeaderboardState > should render HostLeaderboardState 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      >
        <div
          class="main small"
        >
          <button
            class="secondary"
            data-testid="test-next-button-button"
            id="next-button"
            name="next-button"
            type="button"
          >
            Next
            <div
              class="arrow right"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="content center noPadding"
    >
      <div
        class="main"
      >
        <div
          class="title"
        >
          Leaderboard
        </div>
        <div
          class="leaderboard"
        >
          <div
            class="column"
          >
            <div
              class="position"
            >
              <span>
                1
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            ShadowCyborg
          </div>
          <div
            class="column"
          >
            <span
              class="score"
            >
              18456
            </span>
          </div>
          <div
            class="column"
          >
            <div
              class="position"
            >
              <span>
                2
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            Radar
          </div>
          <div
            class="column"
          >
            <span
              class="score"
            >
              18398
            </span>
          </div>
          <div
            class="column"
          >
            <div
              class="position"
            >
              <span>
                3
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            ShadowWhirlwind
          </div>
          <div
            class="column"
          >
            <span
              class="score"
            >
              15492
            </span>
          </div>
          <div
            class="column"
          >
            <div
              class="position"
            >
              <span>
                4
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            WhiskerFox
          </div>
          <div
            class="column"
          >
            <span
              class="score"
            >
              14118
            </span>
          </div>
          <div
            class="column"
          >
            <div
              class="position"
            >
              <span>
                5
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            JollyNimbus
          </div>
          <div
            class="column"
          >
            <span
              class="score"
            >
              13463
            </span>
          </div>
          <div
            class="column"
          >
            <div
              class="position"
            >
              <span>
                6
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            PuddingPop
          </div>
          <div
            class="column"
          >
            <span
              class="score"
            >
              12459
            </span>
          </div>
          <div
            class="column"
          >
            <div
              class="position"
            >
              <span>
                7
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            MysticPine
          </div>
          <div
            class="column"
          >
            <span
              class="score"
            >
              11086
            </span>
          </div>
          <div
            class="column"
          >
            <div
              class="position"
            >
              <span>
                8
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            FrostyBear
          </div>
          <div
            class="column"
          >
            <span
              class="score"
            >
              10361
            </span>
          </div>
          <div
            class="column"
          >
            <div
              class="position"
            >
              <span>
                9
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            Willo
          </div>
          <div
            class="column"
          >
            <span
              class="score"
            >
              9360
            </span>
          </div>
          <div
            class="column"
          >
            <div
              class="position"
            >
              <span>
                10
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            ScarletFlame
          </div>
          <div
            class="column"
          >
            <span
              class="score"
            >
              6723
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
